packages/algolia-experiences/src/render.tsx

```diff
@@ -0,0 +1,252 @@
+/** @jsx h */
+import { getPropertyByPath } from 'instantsearch.js/es/lib/utils';
+import { index, panel } from 'instantsearch.js/es/widgets';
+import { h, Fragment } from 'preact';
+
+import { error } from './util';
+import { widgets } from './widgets';
+
+import type {
+  Child,
+  Configuration,
+  PanelWidget,
+  PanelWidgetTypes,
+  TemplateAttribute,
+  TemplateChild,
+  TemplateText,
+  TemplateWidgetTypes,
+} from './types';
+import type { Widget } from 'instantsearch.js';
+import type { ComponentChildren, JSX } from 'preact';
+
+export function injectStyles() {
+  const style = document.createElement('style');
+  style.dataset.source = 'instantsearch';
+
+  // @TODO: decide if this should be for all columns or only a specific type
+  style.textContent = `
+    .ais-Columns {
+      display: grid;
+      grid-template-columns: minmax(min-content, 200px) 1fr;
+      gap: 1em;
+    }
+  `;
+  document.head.appendChild(style);
+}
+
+export function configToIndex(
+  config: Configuration,
+  elements: Map<string, HTMLElement>
+) {
+  const container = elements.get(config.id);
+  if (!container) {
+    error(`Element with id ${config.id} not found`);
+    return [];
+  }
+
+  return [
+    index({
+      indexName: config.indexName,
+      indexId: config.id,
+    }).addWidgets(
+      config.children.flatMap((child) => childToWidget(child, container))
```

Comment:
it's a flatmap because a single "columns" adds all child widgets as an array, maybe I ended up writing the schema slightly different than in the api though 🤔